File: 백준/Silver/10828. 스택/스택.py
# FIFO
import sys
input = sys.stdin.readline().rstrip()

N = int(input)

slist = []

def push(X) :
    slist.append (X)
    
def pop() :
    if slist == [] :
        print (-1)
        
    else :
        print (slist[-1])
        slist.pop()
        
def size() :
    print (len(slist))
    
def empty() :
    if slist == [] :
        print (1)
        
    else :
        print (0)
        
def top() :
    if slist == [] :
        print (-1)
        
    else :
        print (slist[-1])
        
for i in range (N) :
    input = sys.stdin.readline().rstrip()
    command = input
    
    if 'push' in command :
        push(command.split (' ')[1])
        
    elif command == 'pop' :
        pop()
        
    elif command == 'size' :
        size()
        
    elif command == 'empty' : 
        empty()
        
    elif command == 'top' :
        top()

File: 백준/Silver/10828. 스택/스택.cc
#include <bits/stdc++.h>
using namespace std;

int main() {
    stack<int> s;
    int cnt = 0; 
    
    int N;
    cin >> N;
    
    for (int i = 0; i < N; i++) {
        string command;
        cin >> command;
        
        if (command == "push") { 
            int X;
            cin >> X;
            
            s.push(X);
            cnt++;
        }
        
        else if (command == "pop") {
            if (cnt != 0) {
                cout << s.top() << '\n';
                s.pop();
                cnt--;
            } 
            
            else {
                cout << -1 << '\n';;
            }
        }
        
        else if (command == "size") {
            cout << cnt << '\n';
        }
        
        else if (command == "empty") {
            cout << s.empty() << '\n';
        }
        
        else if (command == "top") {
            if (!s.empty()) { 
                cout << s.top() << '\n';
            } 
            
            else {
                cout << -1 << '\n';
            }
        }
    }
    
    return 0;
}
